### PR TITLE
Fix some -Wfloat-conversion warnings (Pt. 5)

### PR DESCRIPTION
**Pull request by Holzhaus**
We're getting close...

### CODE REVIEW

File: src/effects/builtin/phasereffect.cpp [hunk 1]
```diff
@@ -167,11 +170,11 @@ void PhaserEffect::processChannel(const ChannelHandle& handle,
     }
     // freqSkip is used to calculate the phase independently for each channel,
     // so do not multiply periodSamples by the number of channels.
-    CSAMPLE freqSkip = 1.0 / periodSamples * 2.0 * M_PI;
+    const auto freqSkip = static_cast<CSAMPLE>(1.0 / periodSamples * 2.0 * M_PI);
```
Comment (daschuer) [inline src/effects/builtin/phasereffect.cpp:173]: use kDoublePi? 
I think this can become float math

File: src/effects/builtin/phasereffect.cpp [hunk 2]
```diff
@@ -199,25 +202,27 @@ void PhaserEffect::processChannel(const ChannelHandle& handle,
         right = pInput[i + 1] + tanh(right * feedback);
 
         // For stereo enabled, the channels are out of phase
-        pState->leftPhase = fmodf(pState->leftPhase + freqSkip, 2.0 * M_PI);
-        pState->rightPhase = fmodf(pState->rightPhase + freqSkip + M_PI * stereoCheck, 2.0 * M_PI);
+        pState->leftPhase = fmodf(pState->leftPhase + freqSkip, kDoublePi);
+        pState->rightPhase = fmodf(static_cast<float>(pState->rightPhase +
```
Comment (daschuer) [inline src/effects/builtin/phasereffect.cpp:206]: cast just M_PI?

File: src/engine/enginemaster.cpp [hunk 3]
```diff
@@ -412,9 +412,9 @@ void EngineMaster::process(const int iBufferSize) {
     CSAMPLE pflMixGainInHeadphones = 1;
     CSAMPLE masterMixGainInHeadphones = 0;
     if (masterEnabled) {
-        CSAMPLE cf_val = m_pHeadMix->get();
-        pflMixGainInHeadphones = 0.5 * (-cf_val + 1.);
-        masterMixGainInHeadphones = 0.5 * (cf_val + 1.);
+        const auto cf_val = static_cast<CSAMPLE>(m_pHeadMix->get());
```
Comment (daschuer) [inline src/engine/enginemaster.cpp:415]: static_cast<CSAMPLE_GAIN>

File: src/engine/enginemaster.cpp [hunk 4]
```diff
@@ -710,7 +710,7 @@ void EngineMaster::process(const int iBufferSize) {
         // Balance values
         CSAMPLE balright = 1.;
         CSAMPLE balleft = 1.;
-        CSAMPLE bal = m_pBalance->get();
+        const auto bal = static_cast<CSAMPLE>(m_pBalance->get());
```
Comment (daschuer) [inline src/engine/enginemaster.cpp:713]: static_cast<CSAMPLE_GAIN>

File: src/engine/enginepregain.cpp [hunk 5]
```diff
@@ -64,7 +64,7 @@ void EnginePregain::setSpeedAndScratching(double speed, bool scratching) {
 }
 
 void EnginePregain::process(CSAMPLE* pInOut, const int iBufferSize) {
-    const float fReplayGain = m_pCOReplayGain->get();
+    const auto fReplayGain = static_cast<float>(m_pCOReplayGain->get());
```
Comment (daschuer) [inline src/engine/enginepregain.cpp:67]: static_cast<CSAMPLE_GAIN>

File: src/engine/enginepregain.cpp [hunk 6]
```diff
@@ -64,7 +64,7 @@ void EnginePregain::setSpeedAndScratching(double speed, bool scratching) {
...
     float fReplayGainCorrection;
```
Comment (daschuer) [inline src/engine/enginepregain.cpp:68]: CSAMPLE_GAIN

File: src/engine/enginepregain.cpp [hunk 7]
```diff
@@ -99,8 +99,9 @@ void EnginePregain::process(CSAMPLE* pInOut, const int iBufferSize) {
             if (seconds < kFadeSeconds) {
                 // Fade smoothly
                 double fadeFrac = seconds / kFadeSeconds;
-                fReplayGainCorrection = m_fPrevGain * (1.0 - fadeFrac) +
-                        fadeFrac * fullReplayGainBoost;
+                fReplayGainCorrection =
```
Comment (daschuer) [inline src/engine/enginepregain.cpp:102]: this can become float math

File: src/engine/enginepregain.cpp [hunk 8]
```diff
@@ -140,7 +141,7 @@ void EnginePregain::process(CSAMPLE* pInOut, const int iBufferSize) {
     } else {
         speedGain = math_min(kMaxTotalGainBySpeed / totalGain, speedGain);
```
Comment (daschuer) [inline src/engine/enginepregain.cpp:142]: this can also become float math

**Review by daschuer — CHANGES_REQUESTED** [review]
Thanks :-)

**Comment by Holzhaus** [comment]
Merge?

**Review by daschuer — APPROVED** [review]
LGTM Thank you. 
waiting for CI

**Comment by Holzhaus** [comment]
Merge?

**Pull request merged**